### PR TITLE
Changes to support notary's unify-cryptoservice branch

### PR DESCRIPTION
**Pull request by aaronlehmann**
- Add RSAPKCS1v15Signature signature algorithm (for hardware signing)
- Use data.Key interface more widely, to allow alternative key
  implementations
- Add additional functions to the CryptoService interface

### CODE REVIEW

File: signed/ed25519.go [hunk 1]
```diff
@@ -24,8 +24,9 @@ func (trust *Ed25519) addKey(k *data.PrivateKey) {
 	trust.keys[k.ID()] = k
 }
 
-func (trust *Ed25519) RemoveKey(keyID string) {
+func (trust *Ed25519) DeleteKey(keyID string) error {
```
Comment (diogomonica) [inline signed/ed25519.go]: Why is this called "trust" in the first place?

File: signed/interface.go [hunk 2]
```diff
@@ -20,7 +20,13 @@ type KeyService interface {
...
 	// The role isn't currently used for anything, but it's here to support
 	// future features
-	Create(role string, algorithm data.KeyAlgorithm) (*data.PublicKey, error)
+	Create(role string, algorithm data.KeyAlgorithm) (data.Key, error)
+
+	// GetKey retrieves the public key if present, otherwise it returns nil
+	GetKey(keyID string) data.Key
+
+	// DeleteKey deletes the specified key
+	DeleteKey(keyID string) error
```
Comment (diogomonica) [inline signed/interface.go]: Don't forget to rename this to Remove

File: signed/verifiers.go [hunk 3]
```diff
@@ -116,6 +126,29 @@ func (v RSAPSSVerifier) Verify(key data.Key, sig []byte, msg []byte) error {
 	return verifyPSS(pubKey, digest[:], sig)
 }
 
+// RSAPKCS1v15SVerifier checks RSA PKCS1v15 signatures
+type RSAPKCS1v15Verifier struct{}
```
Comment (diogomonica) [inline signed/verifiers.go:130]: For future reference I made the call of including PKCSv15 in here. The idea is that we don't know what kind of hardware we're going to deal with in the future, and it's better safe than sorry.

I hope we don't regret this decision. Everyone should be strongly encouraged to use PSS.

File: signed/verifiers_test.go [hunk 4]
```diff
@@ -238,7 +349,7 @@ func TestECDSAVerifierWithInvalidSignature(t *testing.T) {
 	assert.NoError(t, err)
 
 	// Modify the signature
-	signedData[0] = []byte("a")[0]
+	signedData[0]++
```
Comment (diogomonica) [inline signed/verifiers_test.go:352]: Ha! This is so much better.

File: signed/ed25519.go (hunk 1 above)
Comment (aaronlehmann) [inline signed/ed25519.go]: It shouldn't be called trust. I'll change it.

**Comment by aaronlehmann** [comment]
Updated to change DeleteKey to RemoveKey and fix the poorly named "trust" variables.

**Pull request merged**